modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -78,21 +77,19 @@ def parse_dacl_or_sacl(acl)
 
       # To decode the ObjectType we need to do another query to CN=Configuration,DC=daforest,DC=com
       # and look at either schemaIDGUID or rightsGUID fields to see if they match this value.
-      next unless ace_body[:flags] && ace_body[:flags][:ace_object_type_present] == 1
+      if (object_type = ace_body[:object_type]) && !(object_type == CERTIFICATE_ENROLLMENT_EXTENDED_RIGHT || object_type == CERTIFICATE_AUTOENROLLMENT_EXTENDED_RIGHT)
+        # If an object type was specified, only process the rest if it is one of these two
+        next
+      end
 
-      object_type = ace_body[:object_type]
+      next if (ace_body.access_mask.protocol & CONTROL_ACCESS) == 0
 
-      if (ace_body.access_mask.protocol & CONTROL_ACCESS) != 0 && (object_type == CERTIFICATE_ENROLLMENT_EXTENDED_RIGHT || object_type == CERTIFICATE_AUTOENROLLMENT_EXTENDED_RIGHT)
-        if ace_string.match(/DENIED/)
-          flag_allowed_to_enroll = false
-        elsif ace_string.match(/ALLOWED/)
-          flag_allowed_to_enroll = true
-          allowed_sids << ace_body[:sid].to_s
-        end
```

Comment:
This logic was causing the last `flag_allowed_to_enroll` value to be returned. That means that if the ACL ended in an ACE denying permissions to some account, the entire thing would be ignored. Now the update only processes SIDs that are allowed to enroll. This does contain an issue (that existed prior to these changes) in that if there is an explicit deny of a SID higher in the ACL, it will be ignored. I'm not sure how common this is. The solution I'd propose if this is a concern would be to create a second array of SIDs that are explicitly denied access and print them both for the user so they at least see the information. We don't really do any evaluation of the SIDs so it doesn't seem worth while to pursue the non-trivial endeavor of processing the ACEs in order to check if a particular SID (such as the currently authenticated user) matches an entry with a result of either allow or deny. This would mostly be complicated because of the nested groups that AD supports.

Anyways for now, let's just list the SIDs that are allowed and check if that list is empty (which is done by the caller instead of checking `flag_allowed_to_enroll)`.